File: src/comun/componente/MemoriaRam.java
package comun.componente;

/**
 *
 * @author Barrionuevo Diego
 */
public class MemoriaRam extends Componente {

    private String banco;//numero o nombre de socalo
    private String tecnologia;
    private String fabricante;
    private String numeroSerie;
    private String tamanioBusDatos;
    private double velocidad;//Hz
    private double capacidad;//kBytes

    public String getBanco() {
        return banco;
    }

    public void setBanco(String banco) {
        this.banco = banco;
    }

    public String getTecnologia() {
        return tecnologia;
    }

    public void setTecnologia(String tecnologia) {
        this.tecnologia = tecnologia;
    }

    public String getFabricante() {
        return fabricante;
    }

    public void setFabricante(String fabricante) {
        this.fabricante = fabricante;
    }

    public String getNumeroSerie() {
        return numeroSerie;
    }

    public void setNumeroSerie(String numeroSerie) {
        this.numeroSerie = numeroSerie;
    }

    public String getTamanioBusDatos() {
        return tamanioBusDatos;
    }

    public void setTamanioBusDatos(String tamanioBusDatos) {
        this.tamanioBusDatos = tamanioBusDatos;
    }

    public double getVelocidad() {
        return velocidad;
    }

    public void setVelocidad(double velocidad) {
        this.velocidad = velocidad;
    }

    public double getCapacidad() {
        return capacidad;
    }

    public void setCapacidad(double capacidad) {
        this.capacidad = capacidad;
    }

}

File: Cliente/Linux/componentes/memoriaram.py

class MemoriaRam():
        
    def __init__(self):        
        self.banco = ""
        self.tecnologias = "" #EPROM, VRAM
        self.fabricante = ""        
        self.numero_serie = ""
        self.tamanio_bus_datos = ""
        self.velocidad = ""
        self.tamanio = ""

    def setid(self, id):
        self.id = id
    
    def setbanco(self, banco):
        self.banco = banco
        
    def setfabricante(self, fabricante):
        self.fabricante = fabricante
    
    def settamanio(self, tamanio):
        """Unidad de medida en Bytes"""
        self.tamanio = tamanio
        
    def setnumeroserie(self,numero_serie):
        self.numero_serie = numero_serie
    
    def settamaniobusdatos(self, tamanio_bus_datos):
        """En Bits"""
        self.tamanio_bus_datos = tamanio_bus_datos
        
    def setvelocidad(self, velocidad):
        """En MHz"""
        self.velocidad = velocidad
        
    def settecnologia(self, tecnologia):
        """EPROM, VRAM"""
        self.tecnologias = tecnologia
    
    def tostr(self):        
        memoria = 'Banco:'+ self.banco + '\n' +'Fabricante:'+self.fabricante+ '\n'+'Tamanio:'+ self.tamanio + '\n' + 'NumeroDeSerie:'+ self.numero_serie+'\n'+'TamanioBusDatos:'+ self.tamanio_bus_datos + '\n' + 'Velocidad:'+ self.velocidad+'\n'+'Tecnologia:'+ self.tecnologias
        return memoria    
